src/components/CommentsList.jsx

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from "react";
+import { fetchComments } from "../api";
+import CommentCard from "./CommentCard";
+import Lottie from "lottie-react";
+import loadingAnimation from "../assets/loadingAnimation.json";
+
+function Comments({ articleId }) {
+  const [comments, setComments] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetchComments(articleId)
+      .then((comments) => {
+        setComments(comments);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        setIsLoading(false);
+        setIsError(true);
+      });
+  }, []);
+
```

Comment:
this file is great because it handles the loading of comments completely separate from the loading of the article. What this means is your article data loads and is displayed while your comments continue to load. On lower network speeds I can keep reading the article while I wait for the comments to load.